src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -6510,6 +6522,24 @@ def set_up_node_resource_group_profile(self, mc: ManagedCluster) -> ManagedClust
         mc.node_resource_group_profile = node_resource_group_profile
         return mc
 
+    def set_up_bootstrap_profile(self, mc: ManagedCluster) -> ManagedCluster:
+        self._ensure_mc(mc)
+
+        bootstrap_artifact_source = self.context.get_bootstrap_artifact_source()
+        bootstrap_container_registry_resource_id = self.context.get_bootstrap_container_registry_resource_id()
+        if bootstrap_artifact_source is not None:
+            if bootstrap_artifact_source != CONST_ARTIFACT_SOURCE_CACHE and bootstrap_container_registry_resource_id:
+                raise MutuallyExclusiveArgumentError(
+                    "Cannot specify --bootstrap-container-registry-resource-id when "
+                    "--bootstrap-artifact-source is not Cache."
+                )
+            if mc.bootstrap_profile is None:
```

Comment:
https://dev.azure.com/azclitools/public/_build/results?buildId=228957&view=logs&j=4e4d232d-230a-55c9-0730-4c1a9c36b9e5&t=6d1bdb99-245b-5117-da18-01d8df6f72c0&l=1174

> >           if mc.bootstrap_profile is None:
> E           AttributeError: 'ManagedCluster' object has no attribute 'bootstrap_profile'

To be compatible with the old profile, you need to ensure that the code can run on the old SDK and the model in the old SDK does not have this attribute. One way to check is something like `hasattr(mc, "bootstrap_profile")`